### PR TITLE
nodejs18 fasttrack for cve-2023-44487

### PR DESCRIPTION
**Pull request by ddstreetmicrosoft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
nodejs18 fasttrack for cve-2023-44487
###### Change Log  

- Change
- Change
- Change
nodejs18 fasttrack for cve-2023-44487
###### Does this affect the toolchain?  

NO

###### Associated issues  

- #xxxx
nodejs18 fasttrack for cve-2023-44487
###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX
nodejs18 fasttrack for cve-2023-44487
###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: SPECS/nodejs/nodejs18.spec [hunk 1]
```diff
@@ -1,12 +1,17 @@
 # Retrieved from 'deps/npm/package.json' inside the sources tarball.
-%define npm_version 9.5.1
+%define npm_version 9.8.1
+
+# Unfortunately our 'filesystem' package conflicts with /usr/lib/debug
+# and /usr/lib/debug/usr from any/all debuginfo packages, so I guess
```
Comment (PawelWMS) [inline SPECS/nodejs/nodejs18.spec:5]: This comment seems incorrect. We publish multiple debuginfo packages providing `/usr/lib/debug` and `/usr/lib/debug/usr` and they install without issues with `filesystem`. Could you clarify the comment a bit?

Comment (PawelWMS) [reply SPECS/nodejs/nodejs18.spec:5]: Also, why was disabling the debuginfo package require now? Earlier versions of `nodejs18` all shipped with their debuginfo packages.

Comment (ddstreetmicrosoft) [reply SPECS/nodejs/nodejs18.spec:5]: check the pipeline failure from the pr into main https://github.com/microsoft/CBL-Mariner/pull/6464

specifically https://dev.azure.com/mariner-org/mariner/_build/results?buildId=438053&view=results

Comment (ddstreetmicrosoft) [reply SPECS/nodejs/nodejs18.spec]: oh, i didn't know we could just ignore pipeline failures; i repushed without the debuginfo disabling commit

Comment (PawelWMS) [reply SPECS/nodejs/nodejs18.spec]: The buddy build pipelines are a bit simplified to be able to perform a faster sanity check. The ones ran in the PR check and in any "official" builds (nightly dev build and monthly + fast-track release builds) represent the proper build environment and also run install/uninstall tests to make sure the new packages don't fail.

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**